sdk/src/wallet/migration/migrate_0.rs

```diff
@@ -169,39 +136,258 @@ trait Convert {
+            transaction_id.copy_from_slice(&self.transaction_id.0);
+            index.copy_from_slice(&self.index.to_le_bytes());
+            write!(f, "{}", prefix_hex::encode(buffer))
+        }
+    }
+
+    string_serde_impl!(OutputId);
+
+    #[derive(Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct OutputMetadata {
+        pub block_id: serde_json::Value,
+        pub output_id: OutputId,
+        pub is_spent: bool,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub milestone_index_spent: Option<u32>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub milestone_timestamp_spent: Option<u32>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub transaction_id_spent: Option<TransactionId>,
+        pub milestone_index_booked: u32,
+        pub milestone_timestamp_booked: u32,
+        pub ledger_index: u32,
+    }
+
+    #[derive(Serialize, Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct OutputMetadataDto {
+        pub block_id: serde_json::Value,
+        pub transaction_id: String,
+        pub output_index: u16,
+        pub is_spent: bool,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub milestone_index_spent: Option<u32>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub milestone_timestamp_spent: Option<u32>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub transaction_id_spent: Option<String>,
+        pub milestone_index_booked: u32,
+        pub milestone_timestamp_booked: u32,
+        pub ledger_index: u32,
+    }
+
+    #[derive(Serialize, Deserialize)]
+    pub enum InclusionState {
+        Pending,
+        Confirmed,
+        Conflicting,
+        UnknownPruned,
+    }
+}
+
 struct ConvertIncomingTransactions;
 impl Convert for ConvertIncomingTransactions {
-    type New = HashMap<TransactionId, Transaction>;
-    type Old = HashMap<TransactionId, (TransactionPayload, Vec<OutputWithMetadataResponse>)>;
+    type New = HashMap<types::TransactionId, types::Transaction>;
+    type Old = HashMap<types::TransactionId, (types::TransactionPayload, Vec<types::OutputWithMetadataResponse>)>;
 
     fn convert(old: Self::Old) -> crate::wallet::Result<Self::New> {
         let mut new = HashMap::new();
         for (tx_id, (tx_payload, inputs)) in old {
-            new.insert(
-                tx_id,
-                build_transaction_from_payload_and_inputs(tx_id, tx_payload, inputs)?,
-            );
+            let types::TransactionEssence::Regular(tx_essence) = &tx_payload.essence;
+            let txn = types::Transaction {
+                network_id: tx_essence.network_id,
+                payload: tx_payload,
+                block_id: inputs
+                    .first()
+                    .map(|i: &types::OutputWithMetadataResponse| i.metadata.block_id.clone()),
+                inclusion_state: types::InclusionState::Confirmed,
```

Comment:
Looks sus

Reply:
Yes, incoming transaction are always confirmed, otherwise we wouldn't get them